### PR TITLE
[pigeon] Improve java generation

### PR DESCRIPTION
**Pull request by matejdro**
Made a bunch of minor improvements to the Java generation part of the pigeon. Those should make usage of pigeon from java side a bit more convenient.

Please see commit messages for full list of improvements.

### CODE REVIEW

File: packages/pigeon/lib/java_generator.dart [hunk 1]
```diff
@@ -14,8 +14,8 @@ const Map<String, String> _javaTypeForDartTypeMap = <String, String>{
   'Int32List': 'int[]',
   'Int64List': 'long[]',
   'Float64List': 'double[]',
-  'List': 'ArrayList',
-  'Map': 'HashMap',
+  'List': 'List<Object>',
+  'Map': 'Map<String, Object>',
```
Comment (gaaclarke) [inline packages/pigeon/lib/java_generator.dart:18]: ArrayList and HashMap are directly lifted from the platform channels documentation as the datatypes the standard message codec is creating.  I don't imagine it's safe or possible just to cast from one to the other:

https://flutter.dev/docs/development/platform-integration/platform-channels

Comment (matejdro) [reply packages/pigeon/lib/java_generator.dart:18]: From what I can see in Flutter's code, they don't care about specific type anywhere:

* https://github.com/flutter/engine/blob/36769af3103819b6e977b71f80761091955fa5d9/shell/platform/android/io/flutter/plugin/common/StandardMessageCodec.java#L260
* https://github.com/flutter/engine/blob/36769af3103819b6e977b71f80761091955fa5d9/shell/platform/android/io/flutter/plugin/common/JSONUtil.java#L71

Comment (matejdro) [reply packages/pigeon/lib/java_generator.dart:18]: Also I don't see scenario where this conversion code would suddenly start caring about type of Lists. They are just iterating through it and storing it into transfer format.

Comment (gaaclarke) [reply packages/pigeon/lib/java_generator.dart:18]: Ok, let me give it a closer look today.  It looks like I'm going to have some time set aside for Pigeon work today.

**Comment by matejdro** [comment]
@gaaclarke What should I do to move this forward? Any changes I need to make to the code?

**Comment by gaaclarke** [comment]
@matejdro no, thank you.  I have an outstanding PR for unit tests for generated java code that I wanted to land first.  If i keep getting pulled away from that I'd like to test it personally at least.  Hopefully I can get around to it this week, thanks for your patience.

**Comment by matejdro** [comment]
@gaaclarke Anything new?

**Review by gaaclarke — APPROVED** [review]
Sorry, I was hoping to have some extra testing capabilities for this PR.  It got stalled longer than I hoped.  This looks good.  I might have to tweak it since I'm running into weird null safety CI issues locally.

**Pull request merged**